stdlib/public/Python/NumpyConversion.swift

```diff
@@ -93,6 +93,13 @@ extension Double : NumpyScalarCompatible {
 
 extension Array : ConvertibleFromNumpyArray
   where Element : NumpyScalarCompatible {
+  /// Creates an `Array` with the same shape and scalars as the specified
+  /// `numpy.ndarray` instance.
+  ///
+  /// - Parameter numpyArray: The `numpy.ndarray` instance to convert.
+  /// - Precondition: The `numpy` Python package must be installed.
+  /// - Precondition: `numpyArray` must be 1-D and have a compatible scalar
```

Comment:
do you want to more explicitly describe when a nil value will be returned? Same for other APIs below.

`Precondition` usually describes cases where the code will lead to undefined behavior (e.g. crash) when the condition is violated.

Reply:
+1. Precondition is definitely not the right term for describing `nil`. We should look at how common failable initializers are documented in the standard library.

Reply:
Thanks for great catch!

A `Returns:` comment is typically used to describe failable initializers and when they return `nil`.

Updated in https://github.com/apple/swift/pull/23069/commits/e7920e443eeb844ac662c91069d9c9025a3c2439, following [this stdlib documentation](https://github.com/apple/swift/blob/948655e850e0a4dd8c0f6d5a86f20f2c462c7303/stdlib/public/core/CString.swift#L137).